File: java-test/java/java-arithmetic/src/main/java/com/leetcode/first/J94.java
package com.leetcode.first;

import java.util.ArrayList;
import java.util.List;

public class J94 {

    public static void main(String[] args) {

    }


    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> list=new ArrayList<>();
        getNode(root,list);
        return list;
    }


    public void getNode(TreeNode node,List<Integer> list){
        if (node==null){
            return;
        }
        getNode(node.left,list);
        list.add(node.val);
        getNode(node.right,list);
    }
}



class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

File: python-test/leetcode/J94.py
class TreeNode:
     def __init__(self, val=0, left=None, right=None):
         self.val = val
         self.left = left
         self.right = right


class Solution:
    def inorderTraversal(self, root: TreeNode) -> list[int]:
        list=[]
        self.getNode(root,list)
        return list

    
    def getNode(self,root:TreeNode,list:list[int]):
        if not root:
            return
        self.getNode(root=root.left,list=list)
        list.append(root.val)
        self.getNode(root=root.right,list=list)
        